src/apps/competitions/unpackers/v1.py

```diff
@@ -0,0 +1,159 @@
+import os
+
+from competitions.unpackers.base_unpacker import BaseUnpacker
+from competitions.unpackers.utils import CompetitionUnpackingException, get_datetime
+
+
+class V15Unpacker(BaseUnpacker):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.competition = {
+            "title": self.competition_yaml.get('title'),
+            "logo": None,
+            "registration_auto_approve": not self.competition_yaml.get('has_registration', True),
+            "docker_image": self.competition_yaml.get('competition_docker_image', 'codalab/codalab-legacy:py3'),
+            "pages": [],
+            "phases": [],
+            "leaderboards": [],
+            # Holding place for phases to reference. Ignored by competition serializer.
+            "tasks": {},
+            "solutions": [],
+        }
+
+    def unpack(self):
+        self._unpack_pages()
+        self._unpack_image()
+        self._unpack_phases()
+        self._unpack_leaderboards()
+
+    def _unpack_pages(self):
+        try:
+            pages = self.competition_yaml['html']
+        except KeyError:
+            raise CompetitionUnpackingException('HTML pages could not be found in the yaml file')
+
+        if 'terms' not in pages:
+            # TODO: SHOULD terms be required? are they required in v1.5? How do we handle this?
```

Comment:
It looks like terms is required, we directly access the key here:

https://github.com/codalab/codalab-competitions/blob/eb8dec26ebea2f3376f5db4a533dd0be14769fa0/codalab/apps/web/models.py#L1804

Reply:
✔️